v-next/hardhat/src/internal/builtin-plugins/network-manager/network-manager.ts

```diff
@@ -71,12 +71,17 @@ export class NetworkManagerImplementation implements NetworkManager {
     const networkConnection = await this.#hookManager.runHandlerChain(
       "network",
       "newConnection",
-      [],
-      async (_context) =>
+      [networkName, chainType, networkConfigOverride],
+      async (
+        _nextContext,
+        nextNetworkName,
+        nextChainType,
+        nextNetworkConfigOverride,
```

Comment:
I modify the `newConnection` hook arguments because we need to be able to modify the network config override in a hook before the network is created.

I also expose network name and chain type in the same manner, even though I don't need it for coverage support specifically, but I think it could be useful for some to know these values ahead of network initialization, too.